core-bundle/src/Resources/contao/classes/DataContainer.php

```diff
@@ -1181,21 +1185,20 @@ protected function panel()
 	}
 
 	/**
-	 * Invalidates a list of cache tags.
+	 * Invalidates the cache tags associated with a given DC
 	 *
 	 * Call this whenever an entry is modified (added, updated, deleted).
 	 *
-	 * @param array $tags
+	 * @param DataContainer $dc
 	 */
-	protected function invalidateCacheTags(array $tags)
+	protected function invalidateCacheTags(DataContainer $dc)
 	{
 		if (!System::getContainer()->has('fos_http_cache.cache_manager'))
 		{
 			return;
 		}
 
-		// Filter empty tags
-		$tags = array_filter(array_unique($tags));
+		$tags = $this->getCacheTags($dc);
```

Comment:
Now that this method is no longer used anywhere else, we might as well remove it and execute the logic here. Then it would also become more obvious how the callback should be named: `oninvalidate_cache_tags_callback` (similar to `onrestore_version_callback`)

What do you think?

Reply:
Nice input! You're right, changed right away: f183f3aca2d318da7587068f56c2f6e2ff0bc651